packages/nrkno-iframe-preview-api/src/iframe-preview.ts

```diff
@@ -41,15 +41,64 @@ export function initPreview<T>(
     sanityClientVersion,
     queryParams = (d) => ({ id: d._id }),
     initialData,
+    debug = false,
   }: PreviewConfig<T>,
   setData: (data: T | undefined) => void | Promise<void>
 ) {
+  const logger = createDebugLogger(debug);
+  const setDataInner = setData;
+  setData = (data: T | undefined) => {
+    logger('Invoking setData with:', data);
+    return setDataInner(data);
+  };
+
+  logger('Initializing iframe preview using config:', {
+    origin,
+    groqQuery,
+    sanityClientVersion,
+    queryParams,
+    initialData,
+    debug,
+  });
+
+  function postMessage(messageType: 'ready' | 'updated', origin: string) {
+    const message = { type: messageType };
+    logger('Send message', message);
+    window.parent?.postMessage(message, origin);
+  }
+
+  function sendGROQ(
+    groqQuery: PreviewConfig<unknown>['groqQuery'],
+    params: Record<string, unknown>,
+    sanityClientVersion = DEFAULT_SANITY_CLIENT_VERSION,
+    origin: string
+  ) {
+    const resolveQuery = typeof groqQuery === 'function' ? groqQuery() : groqQuery;
+    Promise.resolve(resolveQuery)
+      .then((query) => {
+        if (!query || !query.includes('_rev')) {
+          throw new Error(
+            `Invalid groq-message. Query MUST contain _rev as a projected field ala {_rev, ...}. Please refer to the docs.\n
```

Comment:
Kanskje være så vågale at man peker til hvor man finner docs her også?

Reply:
Oooh i like it.